source/jquery.flot.drawSeries.js

```diff
@@ -533,17 +533,32 @@ This plugin is used by flot for drawing lines, plots, bars or area.
                     barLeft = 0;
                     break;
                 case "right":
-                    barLeft = -series.bars.barWidth;
+                    if (typeof series.bars.barWidth === "number") {
```

Comment:
The explicit `typeof` should not be necessary to get the numeric value from either the array or numeric version:
``` js
barLeft = -(series.bars.barWidth[0] || series.bars.barWidth)
```